File: src/main/scala/com/mikea/gae/rx/RxMainImpl.java
package com.mikea.gae.rx;

import com.google.inject.Injector;

import javax.inject.Inject;
import java.util.Set;

/**
 * @author mike.aizatsky@gmail.com
 */
public class RxMainImpl {
    private final Set<RxPipeline> pipelines;
    private final RxImplConfigGen rx;

    @Inject
    public RxMainImpl(Set<RxPipeline> pipelines, RxImplConfigGen configGen) {
        this.pipelines = pipelines;
        this.rx = configGen;
    }

    public static void main(Injector injector, String[] args) {
        injector.getInstance(RxMainImpl.class).run(args);
    }

    private void run(String[] args) {
        for (RxPipeline pipeline : pipelines) {
            pipeline.init(rx);
        }

        rx.generateConfigs();
    }
}

File: src/main/scala/com/mikea/gae/rx/RxMainImpl.scala
package com.mikea.gae.rx

import com.google.inject.Injector
import javax.inject.Inject
import com.mikea.gae.rx.impl.RxImplConfigGen

/**
 * @author mike.aizatsky@gmail.com
 */
object RxMainImpl {
  def main(injector: Injector, args: Array[String]): Unit = {
    injector.getInstance(classOf[RxMainImpl]).run(args)
  }
}

class RxMainImpl @Inject()(pipelines: java.util.Set[RxPipeline], rx: RxImplConfigGen) {
  private def run(args: Array[String]): Unit = {
    import scala.collection.JavaConversions._
    pipelines.map(_.init())
    rx.generateConfigs()
  }
}